src/invoice.rs

```diff
@@ -0,0 +1,88 @@
+use async_graphql::SimpleObject;
+use bson::Uuid;
+use serde::{Deserialize, Serialize};
+
+use crate::http_event_service::OrderEventData;
+
+/// Invoice of an order.
+#[derive(Debug, Serialize, Deserialize, SimpleObject, Clone)]
+pub struct Invoice {
+    pub order_id: Uuid,
+    pub content: String,
```

Comment:
the `content` field does not exist in the domain model currently
I'm not sure if a plain text content field is really helpful. Last time we discussed that the frontend is rendering the invoice using this entity and information from the order.
Also the two addresses and the `dateIssued` and `totalInvoicedAmount` are missing

Reply:
The only thing that the invoice service would add then is the vendor address and an timestamp, everything else is already in the order. Also not sure if it can be theoretically "valid" if the formulation of the invoice is only provided by the frontend, that is why i have opted to generate a string in this service

Reply:
the thing is, as a user, I would expect an invoice in pdf form and not a plain text string
also invoices typically have some kind of "format"
I'm not sure how this is achieved with this text
also there seems to be a lot of stuff that either has to be on an invoice or should be there: https://stripe.com/en-de/resources/more/invoice-requirements#invoice-requirements-what-to-include-on-an-invoice
so if we really want to do this in the backend, I would recommend either generating an html string (which could then be rendered to a pdf by the frontend or another service using pandoc etc.) or directly generating pdfs

Reply:
if you want to generate them in the backend, this might be a good option: https://github.com/pdf-rs/pdf

Reply:
I would prefer generating them here, otherwise this service is a bit redundant. The invoice could otherwise directly be built within the frontend

Reply:
I have now adapted the generation logic, if HTML is preferred, we can convert it with an LLM easily

Reply:
so, I'm really not the business guy here, thus, I think it would be a good idea if @lesser1, @KiLanger, or @delvh could also look at the new generated markdown string and find out if this is a correct invoice
to me it looks fine